### DISCUSSION TITLE
Error "stream reset by client (HTTP_1_1_REQUIRED)"

### DISCUSSION THREAD

**Discussion by flansch** [mitmproxy/mitmproxy#6356] [Q&A]
Hi,

I'm new to mitmproxy and using the console interface on Windows to debug HTTP requests. I now have the problem that sometimes mitmproxy just returns the error message 502 HTTP_1_1_REQUIRED when accessing websites running on IIS configured to use Windows Authentication. This doesn't happen directly after starting mitmproxy but when it has been running for several hours.

What I found out already is, that Internet Information Server doesn't allow HTTP/2 to be used when Windows Authentication is enabled, So I'm wondering why mitmproxy is interfering the communication in a way that leads to this error.

Does anyone have an idea what is going on here?

Thank you and best regards
Daniel

**Comment by mhils** [comment]
If `connection_strategy` is set to `eager` (the default), mitmproxy should not negotiate HTTP/2 with the client if the server does not offer it. Everything else would be a bug. If you can reproduce this somewhat reliably, a debug log with the `proxy_debug` option set to `true` would be splendid.

**Comment by flansch** [comment]
I checked `connection_strategy`, it hasn't been changed from its default value. I also enabled `proxy_debug` and am now in the situation that the error occurs. Unfortunately I didn't find information about the location of the log file, could you please tell me where to look.

**Reply by mhils** [reply]
I recommend you use mitmdump, which logs to stdout if you pass `-vvv`. Be warned that logfiles grow very quickly.

**Comment by flansch** [comment]
I created a log file and attached it. Therefore I started mitmdump, hit refresh in the browser, got the error message and  immediately stopped mitmdump. Look for a request to https://webgis.mydomain.com/magis/. Inside the file I changed the domain name and the Authorization header for a bit of anonymization. 

[mitmdump.log](https://github.com/mitmproxy/mitmproxy/files/12558580/mitmdump.log)
